### PR TITLE
feat: add basic relation field

### PR DESCRIPTION
**Pull request by richardshiue**
The feature is hidden behind a feature flag. To enable it, add `FieldType.Relation` to `_supportedFieldTypes` in `appflowy_flutter/lib/plugins/database/grid/presentation/widgets/header/field_type_list.dart`. 

image

image

image

### Feature Preview

---

#### PR Checklist

- [ ] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/plugins/database/widgets/cell/mobile_grid/mobile_grid_relation_cell.dart [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import 'package:appflowy/mobile/presentation/bottom_sheet/show_mobile_bottom_sheet.dart';
+import 'package:appflowy/plugins/database/widgets/row/cells/cell_container.dart';
+import 'package:appflowy/plugins/database/application/cell/bloc/relation_cell_bloc.dart';
+import 'package:appflowy_popover/appflowy_popover.dart';
+import 'package:flowy_infra_ui/flowy_infra_ui.dart';
+import 'package:flutter/material.dart';
+
+import '../editable_cell_skeleton/relation.dart';
+
+class MobileGridRelationCellSkin extends IEditableRelationCellSkin {
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database/widgets/cell/mobile_grid/mobile_grid_relation_cell.dart:10]: I guess it's better to hide it completely than show "Coming soon", what do you think?

File: frontend/rust-lib/flowy-database2/src/event_handler.rs [hunk 2]
```diff
@@ -920,3 +920,76 @@ pub(crate) async fn update_field_settings_handler(
...
+  // // get the related database
+  // let related_database_id = database_editor
+  //   .get_related_database_id(&cell_id.field_id)
+  //   .await?;
+  // let related_database_editor = manager.get_database(&related_database_id).await?;
+
+  // // validate the changeset contents
+  // related_database_editor
+  //   .validate_row_ids_exist(&params)
+  //   .await?;
```
Comment (Xazin) [inline frontend/rust-lib/flowy-database2/src/event_handler.rs:962]: Remove?

File: frontend/rust-lib/flowy-database2/src/event_handler.rs [hunk 3]
```diff
@@ -920,3 +920,76 @@ pub(crate) async fn update_field_settings_handler(
...
+
+  // update the cell in the database
+  database_editor
+    .update_cell_with_changeset(&view_id, cell_id.row_id, &cell_id.field_id, params)
+    .await?;
+  Ok(())
+}
+
+pub(crate) async fn get_related_row_datas_handler(
```
Comment (Xazin) [inline frontend/rust-lib/flowy-database2/src/event_handler.rs:971]: Data is plural 👀

File: frontend/appflowy_flutter/lib/plugins/database/grid/presentation/widgets/header/type_option/relation.dart [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  State<_DatabaseList> createState() => _DatabaseListState();
+}
+
+class _DatabaseListState extends State<_DatabaseList> {
+  late Future<Either<RepeatedDatabaseDescriptionPB, FlowyError>> future;
+
+  @override
+  void initState() {
+    super.initState();
+    future = DatabaseEventGetDatabases().send();
```
[the remark this answers is not in the document]
Comment (richardshiue) [reply frontend/appflowy_flutter/lib/plugins/database/grid/presentation/widgets/header/type_option/relation.dart:116]: I think this is simple enough for now, as there aren't listeners for individual database changes. I'll revisit this then

File: frontend/rust-lib/flowy-database2/src/event_handler.rs (hunk 3 above)
Comment (richardshiue) [reply frontend/rust-lib/flowy-database2/src/event_handler.rs:1035]: Yeah... I'm thinking row data as a struct, so plural of that xD

File: frontend/appflowy_flutter/lib/plugins/database/widgets/cell/mobile_grid/mobile_grid_relation_cell.dart (hunk 1 above)
Comment (richardshiue) [reply frontend/appflowy_flutter/lib/plugins/database/widgets/cell/mobile_grid/mobile_grid_relation_cell.dart:10]: Relation isn't creatable on mobile yet

File: frontend/rust-lib/event-integration/src/database_event.rs [hunk 5]
```diff
@@ -335,6 +335,16 @@ impl EventIntegrationTest {
...
   }
 
+  pub async fn get_relation_cell(
+    &self,
+    view_id: &str,
+    field_id: &str,
+    row_id: &str,
+  ) -> RelationCellDataPB {
+    let cell = self.get_cell(view_id, row_id, field_id).await;
+    RelationCellDataPB::try_from(Bytes::from(cell.data)).unwrap()
```
Comment (appflowy) [inline frontend/rust-lib/event-integration/src/database_event.rs:345]: Maybe unwrap_or_default()?

**Pull request merged**